src/lib/registration/widgets/registration_code.rb

```diff
@@ -73,13 +74,38 @@ def register
         Storage::InstallationOptions.instance.base_registered = true
       end
 
+      def validate
+        return error(_("System already registered.")) if Registration.is_registered?
```

Comment:
so when system is already registered you basically are stuck here, because you cannot do anything about it.